### ISSUE TITLE
Indexing bug in documentation

### ISSUE THREAD

**Issue by saulpierotti** [limix/bgen-reader-py#51]
**Describe the bug**
Indexing variants using the dask API in the way suggested in the documentation returns a within-partition index instead than an overall index, which is uncorrect when trying to extract genotype probabilities. The probabilities of the wrong variant are loaded.

I noticed a nasty bug in the documentation at [https://bgen-reader.readthedocs.io/en/latest/expectation.html](https://bgen-reader.readthedocs.io/en/latest/expectation.html) and other pages related to the dask-like API. The code example in the documentation is as follows:

```
from bgen_reader import allele_expectation, example_filepath, read_bgen
from texttable import Texttable

filepath = example_filepath("example.32bits.bgen")
bgen = read_bgen(filepath, verbose=False)
variants = bgen["variants"]
samples = bgen["samples"]
genotype = bgen["genotype"]

# This `compute` call will return a pandas data frame,
variant = variants[variants["rsid"] == "RSID_6"].compute()

# from which we retrieve the variant index.
variant_idx = variant.index.values[0]
type = bgen["genotype"]

# Samples is a pandas series, and we retrieve the
# sample index from the sample name.
sample_idx = samples[samples == "sample_005"].index.values[0]
genotype = bgen["genotype"]

# This `compute` call will return a dictionary from which
# we can get the probability matrix the corresponding
# variant.
p = genotype[variant_idx].compute()["probs"][sample_idx]
genotype = bgen["genotype"]

# Allele expectation makes sense for unphased genotypes only,
# which is the case here.
e = allele_expectation(bgen, variant_idx)[sample_idx]
```

The problem is at:

```
variant = variants[variants["rsid"] == "RSID_6"].compute()
variant_idx = variant.index.values[0]
e = allele_expectation(bgen, variant_idx)
```
Using this code `variant_idx` does not contain the overall index but the dask within-partition index, which is incorrect when trying to access the bgen data with the command `e = allele_expectation(bgen, variant_idx)[sample_idx]`.

This can be easily corrected by loading the variant ids in memory with compute() before resetting the index on what is now a pandas series:

```
variant_ids = variants["rsid"].compute().reset_index(drop = True)
variant_idx = variant_ids[variant_ids == "RSID_6"].index.values[0]
e = allele_expectation(bgen, variant_idx)
```

See also relevant page of the dask documentation [here](https://docs.dask.org/en/latest/generated/dask.dataframe.Index.reset_index.html#dask.dataframe.Index.reset_index)

**Comment by CarlKCarlK** [comment]
Thanks for the bug report.

**Comment by horta** [comment]
I think I have fixed it with the above commit!

I will make a release soon. Thanks for the reporting it!

**Comment by horta** [comment]
Ah, the fix consists in setting the index the each partition element according to the global index (proper index). Therefore it doesn't break the interface that relies on the proper usage...

**Comment by horta** [comment]
Released the fix on version `4.0.9`. Let me know how it goes...
